src/common.c

```diff
@@ -86,6 +91,28 @@ int fi_read_file(const char *dir, const char *file, char *buf, size_t size)
 		buf[--len] = '\0';
 
 	return len;
+#else /* _WIN32 */
+	char *path;
+	DWORD len;
+
+	if (asprintf(&path, "%s/%s", dir, file) < 0)
+		return -1;
+
+	HANDLE fd = CreateFileA(path, GENERIC_READ, FILE_SHARE_READ, 0, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, 0);
+	free(path);
+	if (fd == INVALID_HANDLE_VALUE) {
+		return -1;
+	}
+
+	BOOL res = ReadFile(fd, buf, (DWORD)size, &len, 0);
+	CloseHandle(fd);
+
+	if (!res) {
+		return -1;
+	}
+
+	return (int)len;
+#endif /* _WIN32 */
```

Comment:
Apparently common.c isn't quite so common.  Add src/windows/osd.c with the necessary definitions there.  This may require moving some code from common.c into src/unix/osd.c.